utils/src/foreign_field.rs

```diff
@@ -413,33 +449,55 @@ mod tests {
         let big = secp256k1_modulus();
         let bytes = big.to_bytes_be();
         assert_eq!(
-            ForeignElement::<BaseField, 3>::from_be(&bytes),
-            ForeignElement::<BaseField, 3>::from_biguint(big)
+            ForeignElement::<BaseField, LIMB_BITS, 3>::from_be(&bytes),
+            ForeignElement::<BaseField, LIMB_BITS, 3>::from_biguint(big.clone())
+        );
+        assert_eq!(
+            ForeignElement::<BaseField, 15, 18>::from_be(&bytes),
```

Comment:
I would suggest making the tests even more generic, to be sure we do not only test cases we want.
For instance, we could take some random decompositions in two numbers of the field modulus.

I would then get rid of `LIMB_BITS = 88` (as it is now misleading as described above), and define a nested module `kimchi` or smth close to it.